src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2371,8 +2374,14 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                     identifierOrThisOpt != null &&
                     (typeParameterListOpt != null && typeParameterListOpt.ContainsDiagnostics
                       || this.CurrentToken.Kind != SyntaxKind.OpenParenToken && this.CurrentToken.Kind != SyntaxKind.OpenBraceToken && this.CurrentToken.Kind != SyntaxKind.EqualsGreaterThanToken) &&
-                    ReconsiderTypeAsAsyncModifier(ref modifiers, ref type, ref explicitInterfaceOpt, identifierOrThisOpt, typeParameterListOpt))
-                {
+                    ReconsiderTypeAsAsyncModifier(ref modifiers, type, identifierOrThisOpt))
+                {
+                    this.Reset(ref afterTypeResetPoint);
+                    this.Release(ref afterTypeResetPoint);
```

Comment:
>this.Release(ref afterTypeResetPoint); [](start = 20, length = 38)

This doesn't look right. This means that we will release the reset point conditionally. See how reset point is released in  ```EatNullableQualifierIfApplicable```, for example.